cogs/onmyoji.py

```diff
@@ -285,6 +323,376 @@ def shiki_bounty_embed(self, shiki):
+
+    @shard.command(name="notes")
+    async def shard_set_notes(self, ctx, *notes):
+        """
+        Sets the user note for their shard trading database entry.
+        Leave the field blank to disable, otherwise type in a note to set that note.
+        """
+        args = ' '.join(notes)
+        self.shard_load_json()
+        if self.check_trading_status(ctx.author.id) == True:
+            trading_status = "available"
+        else:
+            trading_status = "unavailable"
+        if not args:
+            self.shard_trading_db[str(ctx.author.id)]['notes'] = ''
+            await ctx.send(f"Notes disabled.\nYou are currently {bold(trading_status)} for trading.")
+            self.shard_file_writeout()
+        else:
+            self.shard_trading_db[str(ctx.author.id)]['notes'] = args
+            await ctx.send(f"Your Shard Trading entry Notes have been set to:\n```{args}```\nYou are currently {bold(trading_status)} for trading.")
+            self.shard_file_writeout()
+
+    def check_trading_status(self, user):
+        try:
+            return self.shard_trading_db[str(user)]['status']
+        except KeyError:
+            return True
+
+    @shard.command(name="status")
+    async def shard_set_trading_status(self, ctx, *arg):
+        """
+        Toggles the trading status for the user.
+        If no terms are given, toggles it.
+        Otherwise, if on/off is given as a term, sets it to that state.
+        """
+        self.shard_load_json()
+        try:
+            temp = self.shard_trading_db[str(ctx.author.id)]['status']
+        except KeyError:
+            self.shard_trading_db[str(ctx.author.id)]['status'] = True
+            self.shard_file_writeout();
+            return await ctx.send(f"{ctx.author.mention} is now available to be searched for trading.")
+        if not arg:
+            if self.check_trading_status(ctx.author.id) == True:
+                self.shard_trading_db[str(ctx.author.id)]['status'] = False
+                self.shard_file_writeout();
+                return await ctx.send(f"{ctx.author.mention} is now unavailable to be searched for trading.")
+            else:
+                self.shard_trading_db[str(ctx.author.id)]['status'] = True
+                self.shard_file_writeout();
+                return await ctx.send(f"{ctx.author.mention} is now available to be searched for trading.")
+        else:
+            if arg == "on":
+                self.shard_trading_db[str(ctx.author.id)]['status'] = True
+                self.shard_file_writeout();
+                return await ctx.send(f"{ctx.author.mention} is now available to be searched for trading.")
+            elif arg == "off":
+                self.shard_trading_db[str(ctx.author.id)]['status'] = False
+                self.shard_file_writeout();
+                return await ctx.send(f"{ctx.author.mention} is unavailable to be searched for trading.")
+
+    def compare_shard_db(self, main_user, other_user):
```

Comment:
```suggestion
    def get_shiki_set(user, list_name):
        return set([''.join(i for i in value if not i.isdigit()).strip().lower() for value in self.shard_trading_db[user][list_name]])
        
    def compare_shard_db(self, main_user, other_user):
        user1_have_list = self.get_shiki_set(main_user, 'have')
        user1_need_list = self.get_shiki_set(main_user, 'need')
        user2_have_list = self.get_shiki_set(other_user, 'have')
        user2_need_list = self.get_shiki_set(other_user, 'need')
        you_have_they_need = user1_have_list.intersect(user2_need_list)
        you_need_they_have = user1_need_list.intersect(user2_want_list)
        ...
```

Reply:
jesus you yoinked my code from 30 lines to like 5
im upset that took me some time lol